tests/frontend_selenium/tests/test_dedicate.py

```diff
@@ -0,0 +1,284 @@
+from utils.utils import generate_string,get_seed
+from pages.polka import PolkaPage
+from pages.dedicate import DedicatePage
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+import pytest
+
+#  Time required for the run (12 cases) is approximately 3 minutes.
+
+"""
+  Test Case: TC1138 - Navigate to dedicate node
+  Steps:
+      - Navigate to the dashboard.
+      - Select an account.
+      - Click on Dedicate Node from side menu.
+  Result: User should be navigated to Dedicate Node page.
+"""
+def test_dedicate_page(browser):
+  dedicate_page = DedicatePage(browser)
+  polka_page = PolkaPage(browser)
+  user = generate_string()
+  password = generate_string()
+  dedicate_page.load()
+  polka_page.authenticate()
+  polka_page.import_account(get_seed(),user,password)
+  dedicate_page.navigate(user)
+  assert 'Dedicated Nodes' in browser.page_source
+
+
+"""
+  Test Case: TC997 - Verify the search functionality by valid location or node id
+  Steps:
+      - From the Twin Details Page Click on Dedicated nodes
+      - In the Farms search bar enter the location or the id of the node
+  Test Data: [ ID - Location - case sensitive Location ]
+  Result: The search results should be displayed correctly according to the keywords used.
+"""
+def test_search_by_valid_name_address(browser):
+  dedicate_page = DedicatePage(browser)
+  polka_page = PolkaPage(browser)
+  user = generate_string()
+  password = generate_string()
+  dedicate_page.load()
+  polka_page.authenticate()
+  polka_page.import_account(get_seed(),user,password)
+  dedicate_page.navigate(user)
+  ids = dedicate_page.get_node_id()
+  locations = dedicate_page.get_node_location()
+  for i in range (len(ids)):
+    dedicate_page.search_nodes(ids[i])
+    assert  len(browser.find_elements(By.XPATH, '//*[@id="app"]/div[1]/div[3]/div/div/div[2]/div[1]/table/tbody/tr')) == 1
```

Comment:
the test shouldn't search for elements. this element definition should be on the page module and used here.